xdsl/dialects/builtin.py

```diff
@@ -2179,23 +2253,28 @@ def from_list(
     @staticmethod
     def from_list(
         type: (
-            RankedStructure[AnyFloat | IntegerType | IndexType]
+            RankedStructure[AnyFloat | IntegerType | IndexType | ComplexType]
             | RankedStructure[AnyFloat]
             | RankedStructure[IntegerType]
             | RankedStructure[IndexType]
         ),
-        data: Sequence[int | float] | Sequence[FloatAttr],
+        data: Sequence[int | float | complex] | Sequence[FloatAttr],
     ) -> DenseIntOrFPElementsAttr: ...
 
     @staticmethod
     def from_list(
         type: (
-            RankedStructure[AnyFloat | IntegerType | IndexType]
+            RankedStructure[AnyFloat | IntegerType | IndexType | ComplexType]
```

Comment:
It might be worth rebasing to main at some point, as it seems you are on a version before when `DenseIntOrFPElementAttr` became generic.